cln_lspd.go

```diff
@@ -217,49 +224,59 @@ func OnHtlcAccepted(event *glightning.HtlcAcceptedEvent) (*glightning.HtlcAccept
 	return event.Continue(), err
 }
 
-func clnGetChannel(clientcln *glightning.Lightning, destination string, fundingTxID string) string {
+func clnGetChannel(clientcln *glightning.Lightning, destination string, fundingTxID string) (string, bool) {
 	obj, err := clientcln.ListFunds()
 	if err != nil {
 		log.Printf("clientcln.ListChannels(%v) error: %v", destination, err)
-		return ""
+		return "", false
 	}
 	for _, c := range obj.Channels {
-		log.Printf("getChannel(%v): %v", destination, c.ShortChannelId)
+		log.Printf("getChannel destination: %v, Short channel id: %v, nodeID: %v , FundingTxID:%v ", destination, c.ShortChannelId, c.Id, c.FundingTxId)
 		if c.FundingTxId == fundingTxID && c.Connected {
 			chanid := c.ShortChannelId
-			return chanid
+			return chanid, true
 		}
 	}
 	log.Printf("No channel found: getChannel(%v)", destination)
-	return ""
+	return "", false
 }
 func clnResumeOrCancel(clientcln *glightning.Lightning, destination string, fundingTxID string, paymentHash string, outgoingAmountMsat uint64, riskfactor float32, event *glightning.HtlcAcceptedEvent, channelPoint string) (*glightning.HtlcAcceptedResponse, error) {
 	deadline := time.Now().Add(60 * time.Second)
 
 	for {
-		shortChanID := clnGetChannel(clientcln, destination, fundingTxID)
+		shortChanID, channelOpenedFlag := clnGetChannel(clientcln, destination, fundingTxID)
 
-		if shortChanID != "" {
+		if channelOpenedFlag != false {
 			log.Printf("channel opended successfully chanID: %v)", shortChanID)
 			dest, err := hex.DecodeString(destination)
 			if err != nil {
 				log.Printf("hex.DecodeString(destination) error: %v", err)
 				break
 			}
-			fields := strings.Split(shortChanID, "x")
 			var chanID lnwire.ShortChannelID
 			var temp []int64
-			for i, field := range fields {
-				temp[i], _ = strconv.ParseInt(field, 10, 64)
-			}
-			chanID = lnwire.ShortChannelID{BlockHeight: uint32(temp[0]), TxIndex: uint32(temp[1]), TxPosition: uint16(temp[2])}
-			// we don't get short channel id until 1 conf
-			err = insertChannel(chanID.ToUint64(), channelPoint, dest, time.Now())
-			if err != nil {
-				log.Printf("insertChannel error: %v", err)
+			var channelId uint64
+			if shortChanID == "" {
+				channelId = uint64(0)
+				err = insertChannel(channelId, channelPoint, dest, time.Now())
```

Comment:
This code is the same for empty channel or real channel, can be reused after the if condition.